refactor: Remove IGptDecoderBatched interface

Funatiq opened the pull request:
## Description

- Removed the IGptDecoderBatched interface and its references.
- Moved decoder_batch::Input class to gptDecoderBatchedInput.h for handling input/output in the GPT decoder.
- Updated includes in relevant files to reflect the new structure.
- Adjusted method signatures in GptDecoderBatched to remove interface dependencies.

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- Refactor
  - Reworked batched decoding API with a new input structure and clearer namespace separation.
  - Standardized tensor handle type across batch inputs.
  - Simplified decoder class design by removing legacy interface inheritance and adding accessors.
  - Includes breaking changes for users integrating batched decoding.

- Chores
  - Cleaned up and updated dependencies and includes to match the new API surface.

- Tests
  - Adjusted tests to use the updated decoding API and types.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Funatiq committed:
refactor: Remove IGptDecoderBatched interface

- Removed the IGptDecoderBatched interface and its references.
- Moved decoder_batch::Input class to gptDecoderBatched.h.
- Updated includes in relevant files to reflect the new structure.
- Adjusted method signatures in GptDecoderBatched to remove interface dependencies.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #10241](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10241/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7568](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7568/) completed with status: 'SUCCESS'
Pipeline passed with automatic retried tests. Check the [rerun report](https://urm.nvidia.com/artifactory/sw-tensorrt-generic/llm-artifacts//LLM/main/L0_MergeRequest_PR/7568/test-results/rerun_report.html) for details.

Funatiq committed:
refactor: Update includes for decoder state integration

- Removed unused includes from gptDecoderBatched.h.
- Added decoderState.h include to trtGptModelInflightBatching.cpp and bindings.cpp to support new decoder state functionality.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
refactor: Update includes for gptDecoder integration

- Removed gptDecoder.h include from gptDecoderBatched.h. Used forward declaration instead.
- Added gptDecoder.h include to trtGptModelInflightBatching.cpp, and gptDecoderBatched.cpp to support new include structure.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #10293](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10293/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #10365](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10365/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #10365](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10365/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7663](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7663/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #10384](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10384/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #10384](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10384/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7678](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7678/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #10390](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10390/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7682](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7682/) completed with status: 'FAILURE'

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #10460](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10460/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7741](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7741/) completed with status: 'SUCCESS'
Pipeline passed with automatic retried tests. Check the [rerun report](https://urm.nvidia.com/artifactory/sw-tensorrt-generic/llm-artifacts//LLM/main/L0_MergeRequest_PR/7741/test-results/rerun_report.html) for details.
